pabi_direct_sql/models/analytic.py

```diff
@@ -0,0 +1,203 @@
+        ======================================
+        This should be the top most method which will not call any super(),
+        as such, it must include all of the following logic here!!!
+        - All logic of [account.analytic.line].create()
+        - 4 ORM fileds, _prepare_orm_defaults
+        - Defaults, _prepare_defaults
+        - Related fields, _prepare_related_values
+        - Compute fields (with store=True)
+        - Constraints (that update fields)
+        """
+        key = 'pabi_direct_sql.analytic_create'
+        config_obj = self.env['ir.config_parameter']
+        if str(config_obj.get_param(key)).lower() == 'true':
+            begin = time.time()
+            _logger.info("------- Direct SQL, create analytic line -------")
+            # Main preparation
+            self._prepare_orm_create(vals)
+            self._prepare_orm_defaults(vals)
+            self._prepare_defaults(vals)
+            self._prepare_related_values(vals)
+            # Computed fields functions
+            self._prepare_compute_quarter(vals)
+            # Special constraints methods
+            self._prepare_check_compute_fiscalyear_id(vals)
+            self._prepare_compute_document(vals)
+            # Do SQL Insert
+            _logger.info("vals: %s", str(vals))
+            keys = [key for key, value in vals.items() if value]
+            values = [value for key, value in vals.items() if value]
+            sql = "insert into %s (%s) values (%s) returning id" % (
+                self._table, ', '.join(keys),
+                ', '.join(['%s' for x in range(len(keys))])
+            )
+            self._cr.execute(sql, tuple(values))
+            analytic_line = self.sudo().browse(self._cr.fetchone()[0])
+            diff = time.time() - begin
+            _logger.info("analytic_line: %s, %s", str(analytic_line), diff)
+            return analytic_line
+        else:
+            return super(AccountAnalyticLine, self).create(vals)
+
+    @api.model
+    def _prepare_orm_create(self, vals):
+        # From account_budget_activity.models.analytic.create()
+        if vals.get('account_id', False):
+            Analytic = self.env['account.analytic.account'].sudo()
+            analytic = Analytic.browse(vals['account_id'])
+            if not analytic:
+                analytic = Analytic.create_matched_analytic(self)
+            if analytic:
+                domain = Analytic.get_analytic_search_domain(analytic)
+                vals.update(dict((x[0], x[2]) for x in domain))
+        # Prepare period_id for reporting purposes
+        date = vals.get('date', fields.Date.context_today(self))
+        if date:
+            periods = self.env['account.period'].find(date)
+            period = periods and periods[0] or False
+            vals.update({'period_id': period.id})
+
+    @api.model
+    def _prepare_orm_defaults(self, vals):
+        today = vals.get('write_date', fields.Date.context_today(self))
```

Comment:
```suggestion
        today = vals.get('write_date', fields.Datetime.now())
```
In database keep datetime and
Should be used format 0.## in code before create analytic line?
![Selection_008](https://user-images.githubusercontent.com/20896369/78642688-522ce180-78dd-11ea-9682-d2b208da3445.png)